zap/src/main/java/org/parosproxy/paros/control/MenuFileControl.java

```diff
@@ -595,14 +598,35 @@ public String getDescription() {
                     return;
                 }
                 // Import the context
-                Model.getSingleton().getSession().importContext(file);
-
-                // Show the dialog
-                View.getSingleton()
-                        .showSessionDialog(
-                                Model.getSingleton().getSession(),
-                                Constant.messages.getString("context.list"),
-                                true);
+                Context ctx = Model.getSingleton().getSession().importContext(file, false);
```

Comment:
If we validate the name before importing the context, could we avoid creating the new `importContext(file, validateName)` method? It would mean we validate twice but that should be fine?

Reply:
How would we have the name if we haven’t imported it?

Reply:
Ah, I missed that.
Alternatively, what do you think about accepting a function for validating the name (e.g. using [Consumer](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/function/Consumer.html) as a parameter in `importContext`?

```java
public Context importContext(File file, Consumer<String> validateContextName)
```

and then `MenuFileControl` can pass a lambda with the custom name validation logic.

The body of the existing `importContext(File file)` could be changed to
```java
return importContext(file, this::validateContextName)
```

Reply:
The idea was to read the XML file directly (and validate the name) not call the import method.

Reply:
Okay I could do it that way, it isn't what we discussed a few weeks ago...

Reply:
I think it was what was decided in the end.

Reply:
So am I going to write the new name to the file then import it? Or am I duplicating the import functionality?

Reply:
How about I extract the middle part of the importContext method into a new one called readContext or something like that, which returns the Context based on having parsed the file content. Then it can be used in the menu and locally in Session?

Reply:
https://github.com/zaproxy/zaproxy/pull/7836#discussion_r1200377197

I was thinking adding a `importContext(ZapXmlConfiguration)`, once the UI overrides the name it can import that.

https://github.com/zaproxy/zaproxy/pull/7836#discussion_r1200472424

In that case yo need to pass the context and the file when importing?

Reply:
In the second one it would take the file and return the context. No validation, not adding/setting in Session.

But I’m good with you first idea around passing the xmlconfig.